// backend/src/protos/auth.proto
syntax = "proto3";

service AuthService {
  rpc RegisterUser (RegisterRequest) returns (RegisterResponse);
  rpc LoginUser (LoginRequest) returns (LoginResponse);
}

message RegisterRequest {
  string username = 1;
  string email = 2;
  string password = 3;
}

message RegisterResponse {
  bool success = 1;
  string message = 2;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  bool success = 1;
  string token = 2;
}